src/main/java/com/hotels/bdp/circustrain/bigquery/extraction/container/UpdateTableSchemaPostExtractionAction.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (C) 2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.circustrain.bigquery.extraction.container;
+
+import org.apache.hadoop.hive.metastore.api.Table;
+
+import com.google.cloud.storage.Storage;
+
+import com.hotels.bdp.circustrain.api.CircusTrainException;
+import com.hotels.bdp.circustrain.bigquery.client.HiveTableCache;
+import com.hotels.bdp.circustrain.bigquery.util.AvroConstants;
+import com.hotels.bdp.circustrain.bigquery.util.SchemaExtractor;
+
+public class UpdateTableSchemaPostExtractionAction implements PostExtractionAction {
+
+  private final String databaseName;
+  private final String tableName;
+  private final HiveTableCache cache;
+  private final Storage storage;
+  private final ExtractionUri extractionUri;
+
+  public UpdateTableSchemaPostExtractionAction(
+      String databaseName,
+      String tableName,
+      HiveTableCache cache,
+      Storage storage,
+      ExtractionUri extractionUri) {
+    this.databaseName = databaseName;
+    this.tableName = tableName;
+    this.cache = cache;
+    this.storage = storage;
+    this.extractionUri = extractionUri;
+  }
+
+  @Override
+  public void run() {
+    Table table = cache.get(databaseName, tableName);
+    if (table == null) {
+      throw new CircusTrainException("Unable to find pre-cached table: " + databaseName + "." + tableName);
+    }
+    String schema = SchemaExtractor.getSchemaFromStorage(storage, extractionUri);
+    table.getSd().getSerdeInfo().putToParameters(AvroConstants.SCHEMA_PARAMETER, schema);
```

Comment:
If this also gets run for a partitioned table then I think this might cause problems when run against older data that isn't AVRO. Let's see what your testing shows. If necessary we can create a separate ticket and PR to capture work to make it backwards compatible (or we just have to clearly document that it isn't and what end users need to do, i.e. start from scratch with their replications).

Reply:
If the table is partitioned and all of it is replicated again, then it looks like the replication might be overwriting the table or something else that replaces the data into a good format.

If it's partitioned and contains a subset of the data in non-avro format, and more avro data is added, then it doesn't look good.